### PR TITLE
Fixed project renaming scripts and Makefile

### PR DESCRIPTION
**Pull request by afritzler**
For me on MacOS I needed to make those changes to make the renaming for project/provider work.

### CODE REVIEW

File: hack/rename-project [hunk 1]
```diff
@@ -16,8 +16,8 @@
 
 project_name=$1
 
-sed -i '' "s/gardener/$project_name/g" pkg/sampleprovider/driver.go
-sed -i '' "s/gardener/$project_name/g" pkg/sampleprovider/machineserver.go
-sed -i '' "s/gardener/$project_name/g" app/controller/main.go
+sed -i "s/gardener/$project_name/g" pkg/sampleprovider/driver.go
```
Comment (prashanth26) [inline hack/rename-project:19]: This works with both GNU and BSD versions of sed. 

sed -i '' -e "s/gardener/$project_name/g" pkg/sampleprovider/driver.go

I am making this change in the upcoming PR here - https://github.com/gardener/machine-controller-manager-provider-sampleprovider/pull/4/commits/21ce1759065b743fda197bc923a844ea463a0fa4. Let me know if this command works on your version of sed. if it does, i could make these changes in the upcoming PR maybe?

File: Makefile [hunk 2]
```diff
@@ -24,7 +24,7 @@ rename-provider:
 	@./hack/rename-provider ${PROVIDER_NAME}
 
 .PHONY: rename-project
-rename-provider:
+rename-project:
```
Comment (prashanth26) [inline Makefile:27]: Made this change in the upcoming PR as well

**Review by prashanth26 — COMMENTED** [review]
I can approve and merge these changes with sed changes suggested with making it work with both. If not, i can close this PR and make these changes in the other PR.

File: hack/rename-project (hunk 1 above)
Comment (afritzler) [reply hack/rename-project:19]: I could make this work locally on my side. So you can go ahead an close this PR if you are already working on a fix.

**Comment by prashanth26** [comment]
Closing this PR in favor of - https://github.com/gardener/machine-controller-manager-provider-sampleprovider/pull/4

**Pull request closed without merging**